spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulConfiguration.java

```diff
@@ -179,8 +186,38 @@ public ZuulRouteApplicationContextInitializer zuulRoutesApplicationContextInitia
 		private Map<String, ZuulFilter> filters;
 
 		@Bean
-		public ZuulFilterInitializer zuulFilterInitializer() {
-			return new ZuulFilterInitializer(this.filters);
+		public ZuulFilterInitializer zuulFilterInitializer(
+				CounterFactory counterFactory, TracerFactory tracerFactory) {
+			return new ZuulFilterInitializer(this.filters, counterFactory, tracerFactory);
+		}
+
+	}
+
+	@Configuration
+	protected static class ZuulEmptyMetricsConfiguration {
+
+		@ConditionalOnMissingBean(CounterFactory.class)
+		@Bean
+		public CounterFactory counterFactory() {
+			return new EmptyCounterFactory();
+		}
+
+		@ConditionalOnMissingBean(TracerFactory.class)
+		@Bean
+		public TracerFactory tracerFactory() {
+			return new EmptyTracerFactory();
+		}
+
+	}
+
+	@Configuration
+	protected static class ZuulMetricsConfiguration {
+
+		@ConditionalOnClass(CounterService.class)
```

Comment:
I'd move this to the class and add a conditional on missing class on the empty one as well.